src/callback.rs

```diff
@@ -35,6 +35,30 @@ impl PyCallbackOutput for () {
     const ERR_VALUE: Self = ();
 }
 
+#[doc(hidden)]
+pub trait IntoResult<R> {
+    fn into_result(self) -> R;
+}
```

Comment:
👍 I think this is a good idea. It might take some refactoring to fully remove result-wrapping from the `IntoPyCallbackOutput` trait. IIRC it was very much needed for `#[pyproto]` macros, however maybe there's a way. Or the full separation could wait to 0.18 even if we merge this now.

Reply:
> It might take some refactoring to fully remove result-wrapping from the IntoPyCallbackOutput trait.

I did try doing this, but got a lot of errors from code related to pyproto 😰